### PR TITLE
Add Docker deploy step for main branch

### PR DESCRIPTION
**Pull request by fmrsabino**
- Deploys production docker image to Docker Hub when the `main` branch is updated
- Cache docker layers with `cache-from` and `cache-to`
- Ignore files which are not required for the docker web image

### CODE REVIEW

File: .github/workflows/publish-staging.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          # Target production image from multi-stage build (check Dockerfile)
+          target: python-base
+          tags: gnosispm/safe-config-service:latest
```
Comment (fmrsabino) [inline .github/workflows/publish-staging.yml:35]: @Uxio0 images built on the `main` branch will be published to Docker Hub with the `latest` tag. Let me know if we should change it to something else

**Comment by fmrsabino** [comment]
And `docker/setup-buildx-action` is quite amazing. Caching the layers of the image reduced a 2m14s build down to 44s 🚀

File: .github/workflows/publish-staging.yml (hunk 1 above)
Comment (Uxio0) [reply .github/workflows/publish-staging.yml:35]: That's reasonable

**Comment by Uxio0** [comment]
> And `docker/setup-buildx-action` is quite amazing. Caching the layers of the image reduced a 2m14s build down to 44s 🚀

I need to try that!

**Review by Uxio0 — APPROVED** [review]

File: .github/workflows/publish-staging.yml (hunk 1 above)
Comment (fmrsabino) [reply .github/workflows/publish-staging.yml]: Actually just synced with @giacomolicari and this was changed to `staging` to be in line with the other services 👍

Comment (Uxio0) [reply .github/workflows/publish-staging.yml]: Makes sense. But they are just tags, you can even push both

**Pull request merged**